build/teamcity/cockroach/nightlies/roachtest_compile_component.sh

```diff
@@ -83,7 +83,7 @@ case "$component" in
     ;;
   roachtest)
     # Roachtest binary.
-    bazel_args=(//pkg/cmd/roachtest)
+    bazel_args=(//pkg/cmd/roachtest --crdb_test)
```

Comment:
What benefit does this give exactly?

Reply:
Stealing from what [Stan said:](https://github.com/cockroachdb/cockroach/pull/113113)

> Note, to fully support interop between a roachtest runner, (executing the roachtest binary), and a node, executing the cockroach binary compiled with assertions (i.e., crdb_test), the roachtest binary must also be compiled with assertions to establish the same serialization on the wire. E.g., KVNemesisSeq is used on the wire only under crdb_test builds. The corresponding (sequence) Container type is conditional on the binary being compiled under crdb_test. (See pkg/kv/kvpb/api.proto) Thus, a request serialized from the roachtest runner is compatible with the cockroach binary iff both binaries are built under crdb_test.

The solution here was to just compile all roachtest binaries with crdb_test since it touches very little code, rather than try and figure out a way to have team city selectively use binaries based on what the cockroach binary is compiled with.